src/bootstrap/test.rs

```diff
@@ -514,6 +514,44 @@ impl Step for RustdocJS {
     }
 }
 
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub struct RustdocUi {
+    pub host: Interned<String>,
+    pub target: Interned<String>,
+    pub compiler: Compiler,
+}
+
+impl Step for RustdocUi {
+    type Output = ();
+    const DEFAULT: bool = true;
+    const ONLY_HOSTS: bool = true;
+
+    fn should_run(run: ShouldRun) -> ShouldRun {
+        run.path("src/test/rustdoc-ui")
+    }
+
+    fn make_run(run: RunConfig) {
+        let compiler = run.builder.compiler(run.builder.top_stage, run.host);
+        run.builder.ensure(tool::Rustdoc {
```

Comment:
No need for this as `builder.rustdoc` below will take care of it for you.

Reply:
Where?

Reply:
`cmd.arg("--rustdoc-path").arg(builder.rustdoc(compiler.host));` on what might be 898 -- that call to `rustdoc` will auto-ensure it for you, so you shouldn't need this.